src/main/java/org/testng/TestNG.java

```diff
@@ -875,9 +875,7 @@ private void initializeCommandLineSuitesGroups() {
     }
   }
   private void addReporter(Class<? extends IReporter> r) {
-    if (! m_reporters.contains(r)) {
-      m_reporters.add(ClassHelper.newInstance(r));
-    }
+    m_reporters.add(ClassHelper.newInstance(r));
```

Comment:
By the way, `addReporter` is marked as private. The only place that uses it is in `initializeDefaultListeners`. For public interface, the consumer needs to do it via command line interface or use `addListener` which does not check any of this.